### PR TITLE
tiny: Return fmt.Errorf instead of duplicated error messages

### PR DESCRIPTION
**Pull request by nak3**

### CODE REVIEW

File: client/alloc_runner.go [hunk 1]
```diff
@@ -204,7 +204,7 @@ func (r *AllocRunner) saveAllocRunnerState() error {
 func (r *AllocRunner) saveTaskRunnerState(tr *TaskRunner) error {
 	var err error
```
Comment (diptanu) [inline client/alloc_runner.go]: Remove the var declaration

File: client/alloc_runner.go [hunk 2]
```diff
@@ -204,7 +204,7 @@ func (r *AllocRunner) saveAllocRunnerState() error {
...
 	if err = tr.SaveState(); err != nil {
-		r.logger.Printf("[ERR] client: failed to save state for alloc %s task '%s': %v",
+		return fmt.Errorf("client: failed to save state for alloc %s task '%s': %v",
 			r.alloc.ID, tr.task.Name, err)
 	}
 	return err
```
Comment (diptanu) [inline client/alloc_runner.go]: `return nil`

Comment (diptanu) [inline client/alloc_runner.go]: `return nil`

File: client/alloc_runner.go (hunk 1 above)
Comment (diptanu) [inline client/alloc_runner.go]: remove the var

File: client/client.go [hunk 3]
```diff
@@ -837,7 +836,7 @@ func (c *Client) registerNode() error {
 	err := c.RPC("Node.Register", &req, &resp)
 	if err != nil {
```
Comment (diptanu) [inline client/client.go]: `if err := c.RPC("Node.Register", &req, &resp)`

File: client/client.go [hunk 4]
```diff
@@ -870,8 +869,7 @@ func (c *Client) updateNodeStatus() error {
 	var resp structs.NodeUpdateResponse
 	err := c.RPC("Node.UpdateStatus", &req, &resp)
 	if err != nil {
```
Comment (diptanu) [inline client/client.go]: Same here

**Comment by diptanu** [comment]
Thanks! Left a few comments.

**Comment by nak3** [comment]
Thank you. I updated them.

File: client/alloc_runner.go [hunk 5]
```diff
@@ -202,12 +202,11 @@ func (r *AllocRunner) saveAllocRunnerState() error {
 }
 
 func (r *AllocRunner) saveTaskRunnerState(tr *TaskRunner) error {
-	var err error
-	if err = tr.SaveState(); err != nil {
-		r.logger.Printf("[ERR] client: failed to save state for alloc %s task '%s': %v",
+	if err := tr.SaveState(); err != nil {
+		return fmt.Errorf("client: failed to save state for alloc %s task '%s': %v",
```
Comment (diptanu) [inline client/alloc_runner.go]: @nak3  we use the format `subsytem: message` only when we are printing the log message. When we create an error, we simply use the error message so in this case the statement would be `fmt.Errorf("failed to save state for alloc %s task '%s': %v",r.alloc.ID, tr.task.Name, err)

I am going to merge this commit and fix this myself.

**Comment by diptanu** [comment]
@nak3 Thanks! Merging this PR, there is one small issue related to formatting of the error messages, I am going to fix them myself.

**Pull request merged**